src/vantage6_strongaya_general/miscellaneous.py

```diff
@@ -0,0 +1,665 @@
+
+    Warnings
+        Setting the 'strict' parameter to False can use invalid data if adhering to a standardised scoring system
+        and consequently skew the output.
+
+    Args:
+        df (pd.DataFrame): The DataFrame to be modified.
+        variables (List[str]): List of variable names to sum.
+        new_variable_name (str): Name of the new variable to store the sum.
+        check_range (tuple[float, float], optional): Expected range for individual items (min, max).
+        required_items (int, optional): Number of items required for valid score calculation.
+        strict (bool): If True, raises errors for invalid data. If False, logs warnings (default: True).
+
+    Returns:
+        pd.DataFrame: The DataFrame with the new summed variable.
+    """
+    # Validate input ranges if specified
+    if check_range is not None:
+        min_val, max_val = check_range
+        for var in variables:
+            invalid_values = ~df[var].between(min_val, max_val)
+            if invalid_values.any():
+                error_msg = (
+                    f"Values in {var} outside expected range [{min_val}, {max_val}]. "
+                    f"Found invalid values in rows: {df.index[invalid_values].tolist()}"
+                )
+                if strict:
+                    raise DataError(error_msg)
+                else:
+                    safe_log("warn", error_msg)
+
+    # Check for missing values
+    missing_count = df[variables].isnull().sum(axis=1)
+    if required_items is not None:
+        invalid_rows = (len(variables) - missing_count) < required_items
+        if invalid_rows.any():
+            error_msg = (
+                f"Insufficient valid responses for {new_variable_name}. "
+                f"Required: {required_items}, Found invalid rows: {df.index[invalid_rows].tolist()}"
+            )
+            if strict:
+                raise DataError(error_msg)
+            else:
+                safe_log("warn", error_msg)
+
+    # Calculate sum
+    safe_log("info", f"Summing variables {variables} into '{new_variable_name}'")
+    df[new_variable_name] = df[variables].sum(axis=1)
+
+    return df
+
+
+def check_variable_availability(df: pd.DataFrame, variables: List[str]) -> bool:
+    """
+    Check if all specified variables are present in the DataFrame.
+
+    Args:
+        df (pd.DataFrame): The DataFrame to check.
+        variables (List[str]): List of variable names to check for.
+
+    Returns:
+        bool: True if all variables are present, False otherwise.
+    """
+    missing_vars = [var for var in variables if var not in df.columns]
+    if missing_vars:
+        safe_log(
+            "error",
+            f"Missing variables in DataFrame: {', '.join(missing_vars)}",
+        )
+        raise UserInputError(
+            "Not all specified variables were found in the working copy of the DataFrame, "
+            "ensure that spelling of input parameters is correct."
+        )
+    return True
```

Comment:
Would be nice to have this in something like `vantage6.algorithm.tools.data_checks`. Let's discuss if/how to create PR(s) to do that